pallets/storage-provider/benchmarks/src/mock.rs

```diff
@@ -0,0 +1,191 @@
+#![cfg(test)]
+
+extern crate alloc;
+
+use std::sync::Arc;
+
+use frame_support::{
+    derive_impl, pallet_prelude::ConstU32, parameter_types, sp_runtime::BoundedVec, PalletId,
+};
+use frame_system::pallet_prelude::BlockNumberFor;
+use primitives::PEER_ID_MAX_BYTES;
+use sp_arithmetic::traits::Zero;
+use sp_keystore::{testing::MemoryKeystore, KeystoreExt};
+use sp_runtime::{
+    traits::{IdentifyAccount, IdentityLookup, Verify},
+    BuildStorage, MultiSignature,
+};
+
+type Block = frame_system::mocking::MockBlock<Test>;
+type BlockNumber = u64;
+
+const MINUTES: BlockNumber = 1;
+
+#[frame_support::runtime]
+mod runtime {
+	#[runtime::runtime]
+	#[runtime::derive(
+		RuntimeCall,
+		RuntimeEvent,
+		RuntimeError,
+		RuntimeOrigin,
+		RuntimeFreezeReason,
+		RuntimeHoldReason,
+		RuntimeSlashReason,
+		RuntimeLockId,
+		RuntimeTask
+	)]
+	pub struct Test;
+
+    #[runtime::pallet_index(0)]
+	pub type System = frame_system::Pallet<Test>;
+
+    #[runtime::pallet_index(1)]
+	pub type Balances = pallet_balances::Pallet<Test>;
+
+    #[runtime::pallet_index(2)]
+	pub type StorageProvider = pallet_storage_provider::Pallet<Test>;
+
+    #[runtime::pallet_index(3)]
+	pub type Market = pallet_market::Pallet<Test>;
+}
+
+pub type Signature = MultiSignature;
+pub type AccountPublic = <Signature as Verify>::Signer;
+pub type AccountId = <AccountPublic as IdentifyAccount>::AccountId;
+
+#[derive_impl(frame_system::config_preludes::TestDefaultConfig)]
+impl frame_system::Config for Test {
+    type Block = Block;
+    type AccountData = pallet_balances::AccountData<u64>;
+    type AccountId = AccountId;
+    type Lookup = IdentityLookup<Self::AccountId>;
+}
+
+#[derive_impl(pallet_balances::config_preludes::TestDefaultConfig)]
+impl pallet_balances::Config for Test {
+    type AccountStore = System;
+}
+
+impl pallet_market::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type PalletId = MarketPalletId;
+    type WeightInfo = ();
+
+    type Currency = Balances;
+    type OffchainSignature = Signature;
+    type OffchainPublic = AccountPublic;
+    type StorageProviderValidation = StorageProvider;
+    type MaxDeals = ConstU32<500>;
+    type MinDealDuration = MinDealDuration;
+    type MaxDealDuration = MaxDealDuration;
+    type MaxDealsPerBlock = ConstU32<500>;
+}
+
+parameter_types! {
+    // Storage Provider Pallet
+    pub const WPoStPeriodDeadlines: u64 = 10;
+    pub const WPoStProvingPeriod: BlockNumber = 40 * MINUTES;
+    pub const WPoStChallengeWindow: BlockNumber = 4 * MINUTES;
```

Comment:
I'm not sure whether those params are affecting benchmarks, however we should leave a note that those are `testnet` parameters and link to the source of them.